### DISCUSSION TITLE
Case insensitive paths?

### DISCUSSION THREAD

**Discussion by ghost** [counterfact/api-simulator#537] [Q&A] [answered]
Can I tell Counterfact to be case-insensitive for the paths? Our OpenApi doc might have `/SomeEndpoint` for a path, but our Javascript client code uses `/someEndpoint` with a lower-case first letter. I can see why Counterfact uses upper first character for the path since that's what the OpenApi doc uses, but can we tell it to ignore case when processing inbound requests?

**Comment by ghost** [comment]
I think I can change `server/registry.js` to handle it. If it works I'll submit a PR.

**Comment by pmcelhaney** [comment]
Sounds good. In the general case, I think it's better for Counterfact to be forgiving. The generated code should preserve case but mapping a URL to a file should be case-insensitive. 

If someone is working with a case-sensitive server they may want Counterfact to behave the same (to prevent erroneous client code that works in Counterfact but not the real thing). But we can make that a setting later. The default should be forgiving.

**Comment by pmcelhaney** [comment, marked as answer]
This will be in the next release. https://github.com/pmcelhaney/counterfact/pull/544
